### PR TITLE
*: escape batch split keys

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
escape batch split keys and only print out the first and last split key.

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No need

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

File: src/pd/pd.rs [hunk 1]
```diff
@@ -132,9 +133,11 @@ impl Display for Task {
                 ..
             } => write!(
                 f,
-                "ask split region {} with keys {:?}",
+                "ask split region {} with {} keys range from {:?} to {:?}",
                 region.get_id(),
-                split_keys
+                split_keys.len(),
+                split_keys.first().as_ref().map(|k| escape(k)),
+                split_keys.last().as_ref().map(|k| escape(k))
```
Comment (overvenus) [inline src/pd/pd.rs]: How about ?

```
for k in keys {
  write!(f, "{}" escape(k));
}
```

Comment (siddontang) [reply src/pd/pd.rs]: em, may output too many keys.

**Comment by siddontang** [comment]
how about we only have one split key? seem the output will be duplicated.

File: src/raftstore/store/msg.rs [hunk 2]
```diff
@@ -244,7 +244,14 @@ impl fmt::Debug for Msg {
                 ref region_id,
                 ref split_keys,
                 ..
-            } => write!(fmt, "Split region {} at key {:?}", region_id, split_keys),
+            } => write!(
+                fmt,
+                "Split region {} with {} keys range from {:?} to from {:?}",
+                region_id,
+                split_keys.len(),
```
Comment (BusyJay) [inline src/raftstore/store/msg.rs]: Can you make it reusable?

Comment (Connor1996) [reply src/raftstore/store/msg.rs]: reuse what?

File: src/raftstore/store/util.rs [hunk 3]
```diff
@@ -764,6 +765,24 @@ impl Engines {
     }
 }
 
+pub fn format_split_info(region_id: u64, split_keys: &[Vec<u8>]) -> String {
```
Comment (BusyJay) [inline src/raftstore/store/util.rs]: Why not introduce a struct that formats the comprehensive information about a `&[Vec<u8>]`? So you don't need to allocate a string for it. You may need to investigate `Display` trait.

File: src/raftstore/store/util.rs [hunk 4]
```diff
@@ -764,6 +765,24 @@ impl Engines {
...
+    if split_keys.len() == 1 {
+        format!(
+            "{} with key {:?}",
+            region_id,
+            split_keys.first().as_ref().map(|k| escape(k)),
+        )
+    } else {
+        format!(
+            "{} with {} keys range from {:?} to {:?}",
```
Comment (BusyJay) [inline src/raftstore/store/util.rs]: `escape` displays the content in debug format already as it *escapes* the invisible chars. Better use `"{}"`, note that `"` is necessary as `escape` doesn't add it for you.

**Comment by Connor1996** [comment]
display example
```
region 1 with 2 keys range from 
"zt\200\000\000\000\000\000\000\377\007\000\000\000\000\000\000\000\370" to 
"zt\200\000\000\000\000\000\000\377\013\000\000\000\000\000\000\000\370"
```

PTAL @BusyJay @zhangjinpeng1987

**Pull request merged**